src/lib/molecules/Copy/Copy.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import classnames from 'classnames';
+import { callAfterDelay } from 'utils';
+import { useHover } from 'hooks';
+import { Icon, Tooltip } from '../../../index';
+
+import './copy.scss';
+
+interface ICopyProps {
+    /**
+     * Size of the icon: 'small', 'medium', 'big',
+     */
+    size?: 'small' | 'medium' | 'big';
+    /**
+     * Additional CSS class name(s) to apply to the icon for styling
+     */
+    className?: string;
+    /**
+     * Reference to the content you want to copy
+     */
+    contentRef?: React.RefObject<HTMLElement>;
+    /**
+     * Determines whether the copy icon is displayed only on hover.
+     */
+    displayOnHover?: boolean;
+    /**
+     * Tooltip text to display when the copy action is available.
+     */
+    copyTooltipText?: string;
+    /**
+     * Tooltip text to display when the copy action has been performed.
+     */
+    copiedTooltipText?: string;
+    /**
+     * The value to copy. If specified, this will be copied instead of the content referenced by contentRef.
+     */
+    value?: string;
+    /**
+     * Determines the visibility state of the copy icon. If true, the icon is visible regardless of hover or controlled visibility.
+     */
+    isVisible?: boolean;
+}
+
+const Copy: React.FC<ICopyProps> = ({
+    size,
+    value,
+    isVisible,
+    className,
+    contentRef,
+    displayOnHover,
+    copyTooltipText,
+    copiedTooltipText
+}) => {
+    const [isCopied, setIsCopied] = useState(false);
+    const [isControlledVisibility, setIsControlledVisibility] = useState(false);
+    const isHovered: boolean = contentRef ? useHover(contentRef) : false;
+
+    useEffect(() => setIsControlledVisibility(isVisible !== undefined), [isVisible]);
+
+    const copyContent = useCallback(() => {
```

Comment:
[Here an article about Return Early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8). 
Please read it and if you agree let's refactor code following that pattern.